src/reference/config/bindings/binding-asyncapi.md

```diff
@@ -0,0 +1,383 @@
+          - api-id: my-mqtt-api-spec
+            operation-id: sendEvents
+        exit: asyncapi_client
+        with:
+          api-id: my-kafka-api-spec
+          operation-id: toSensorData
+      - when:
+          - api-id: my-mqtt-api-spec
+            operation-id: receiveEvents
+        exit: asyncapi_client
+        with:
+          api-id: my-kafka-api-spec
+          operation-id: onSensorData
+  asyncapi_client:
+    type: asyncapi
+    kind: client
+    options:
+      specs:
+        my-kafka-api-spec: kafka/asyncapi.yaml
+      tcp:
+        host: localhost
+        port:
+          - 9092
+```
+
+## Summary
+
+Defines a binding with `asyncapi` spec, with `server` or `proxy` or `client` behavior.
+
+The `server` kind `asyncapi` binding creates composite of `tcp`, `tls`, and `mqtt` or `http` bindings with server kind and adapts MQTT/HTTP streams to AsyncAPI streams.
+
+The `proxy` kind `asyncapi` binding creates composite of `mqtt-kafka` binding with proxy kind mapping MQTT streams to Kafka streams.
+
+The `client` kind `asyncapi` binding creates composite of `kafka` or `mqtt` or `http`, and `tls`, `tcp` bindings with client kind and adapts AsyncAPI streams to Kafka/MQTT/HTTP streams.
+
+## Configuration
+
+:::: note Properties
+
+- [kind\*](#kind)
+- [options](#options)
+  - [options.spec](#options-spec)
+  - [options.tcp](#options-tcp)
+    - [tpc.host](#tpc-host)
+    - [tcp.port](#tcp-port)
+  - [options.http](#options-http)
+    - [http.authorization](#http-authorization)
+    - [authorization.credentials](#authorization-credentials)
+    - [credentials.cookies](#credentials-cookies)
+    - [credentials.headers](#credentials-headers)
+    - [credentials.query](#credentials-query)
+  - [options.tls](#options-tls)
+    - [tls.version](#tls-version)
+    - [tls.keys](#tls-keys)
+    - [tls.trust](#tls-trust)
+    - [tls.signers](#tls-signers)
+    - [tls.trustcacerts](#tls-trustcacerts)
+    - [tls.sni\*](#tls-sni)
+    - [tls.alpn](#tls-alpn)
+    - [tls.mutual](#tls-mutual)
+- [mqtt\_kafka](#mqtt-kafka)
+  - [mqtt\_kafka.channels](#mqtt-kafka-channels)
+    - [channels.sessions](#channels-sessions)
+    - [channels.retained](#channels-retained)
+    - [channels.messages](#channels-messages)
+- [routes\[\].when](#routes-when)
+  - [when\[\].api-id](#when-api-id)
+  - [when\[\].operation-id](#when-operation-id)
+- [routes\[\].exit\*](#routes-exit)
+- [routes\[\].with](#routes-with)
+  - [with.api-id](#with-api-id)
+  - [with.operation-id](#with-operation-id)
+- [exit](#exit)
+
+::: right
+\* required
+:::
+
+::::
+
+### kind\*
+
+> `enum` [ "client", "proxy", "server" ]
+
+Behave as a `asyncapi` `client` or `proxy` or `server`.
+
+```yaml
+kind: server
+```
+
+### options
+
+> `object`
+
+`asyncapi`-specific options.
+
+```yaml
+options:
+  spec:
+    my-asyncapi-spec: spec/asyncapi.yaml
+```
+
+#### options.spec
+
+> `map` of `name: value` properties
+
+AsyncAPI spec definition filename mapped by a unique API spec identifier.
```

Comment:
This is the terminology I landed on where we call out the file name and API spec identifier. Then later when this is referenced in the `api-id` param I use "spec identifier" again for symmetry and to relate with the key name. 

I also have updated the type to correlate with other `name: value` mapping that we do in things like metadata and headers since the schemas are the same. 

```
> `map` of `name: value` properties
```

@jfallows

Reply:
@vordimous LGTM. 👍